.github/workflows/build-test.yml

```diff
@@ -33,32 +59,79 @@ jobs:
         -v $PWD:/build \
         -w /build \
         $BASE_IMAGE:latest-ubuntu-20.04 \
-        /bin/bash -c 'export KERNEL_VERSION=$(ls --indicator-style=none /lib/modules/); echo KERNEL_VERSION=$KERNEL_VERSION; cargo build && cargo build --examples'
+        /bin/bash -c 'cargo clean && export KERNEL_VERSION=$(ls --indicator-style=none /lib/modules/) && echo KERNEL_VERSION=$KERNEL_VERSION && cargo build && cargo build --examples'
+
+  ubuntu-1804-build-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build host info
+      run: |
+        uname -a
+        cat /etc/os-release
+    - name: Initialize git submodules
+      run: |
+        git submodule update --init --recursive
+    - name: Run RedBPF build with kernel v4.19 on ubuntu-18.04 container
+      run: |
+        docker run --privileged \
+        -v $PWD:/build \
+        -w /build \
+        $BASE_IMAGE:latest-ubuntu-18.04 \
+        /bin/bash -c 'cargo clean && export KERNEL_VERSION=$(ls --indicator-style=none /lib/modules/) && echo KERNEL_VERSION=$KERNEL_VERSION && cargo build && cargo build --examples'
+
+  fedora-34-build-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build host info
+      run: |
+        uname -a
+        cat /etc/os-release
+    - name: Initialize git submodules
+      run: |
+        git submodule update --init --recursive
+    - name: Run RedBPF build with kernel headers on fedora-34 container
+      # Run 'make prepare' to prevent generated/autoconf.h from being removed
+      # when 'make -qp' is executed by bpf_sys::headers::build_kernel_version
+      run: |
+        docker run --privileged \
+        -v $PWD:/build \
+        -w /build \
+        $BASE_IMAGE:latest-fedora-34 \
+        /bin/bash -c 'make -C /lib/modules/*/build prepare; cargo clean && export KERNEL_VERSION=$(ls --indicator-style=none /lib/modules/) && echo KERNEL_VERSION=$KERNEL_VERSION && cargo build && cargo build --examples'
+    - name: Run RedBPF build with vmlinux on fedora-34 container
+      run: |
+        docker run --privileged \
+        -v $PWD:/build \
+        -w /build \
+        $BASE_IMAGE:latest-fedora-34 \
+        /bin/bash -c 'cargo clean && /lib/modules/*/build/scripts/extract-vmlinux /lib/modules/*/vmlinuz > /boot/vmlinux && export REDBPF_VMLINUX=/boot/vmlinux && cargo build && cargo build --examples'
 
-  fedora-35-build-test:
```

Comment:
Why don't we do Fedora 35 builds anymore?

Reply:
fedora 35 is not released yet. it is in the development phase.